src/main/java/domain/AttendanceBook.java

```diff
@@ -0,0 +1,118 @@
+public class AttendanceBook {
+    private final HashSet<Attendance> attendances; // key: 몇 일, value: 출석 시간
+
+    public AttendanceBook(int year, int month) {
+        Month customMonth = Month.of(month);
+
+        attendances = new HashSet<>();
+        for (int day = 1; day <= customMonth.getLastDay(); day++) {
+            if (!customMonth.isHoliday(day)) {
+                LocalDate date = LocalDate.of(year, month, day);
+                attendances.add(Attendance.empty(date));
+            }
+        }
+    }
+
+    public Attendance create(LocalDate date, LocalTime time) {
+        final boolean isExist = attendances.stream()
+                .anyMatch(attendance -> attendance.getDate().isEqual(date) && !attendance.isEmpty());
+        if (isExist) {
+            throw new DuplicateAttendanceException();
+        }
+        Attendance attendance = Attendance.of(date, time);
+        addAttendance(attendance);
+        return attendance;
+    }
+
+    public Attendance findAttendanceByDate(LocalDate date) {
+        return attendances.stream()
+                .filter(attendance -> attendance.getDate().isEqual(date))
+                .findFirst()
+                .orElse(Attendance.empty(date));
+    }
+
+    public void replace(LocalDate date, LocalTime time) {
+        Attendance oldAttendance = attendances.stream()
+                .filter(attendance -> attendance.getDate().isEqual(date))
+                .findFirst()
+                .orElseThrow(AttendanceNotExistException::new);
+        Attendance modifiedAttendance = oldAttendance.modify(time);
+        addAttendance(modifiedAttendance);
+    }
+
+    public List<Attendance> getAllAttendances(final int limitDay) {
+        return attendances.stream()
+                .sorted(Comparator.comparing(Attendance::getDate))
+                .filter(attendance -> attendance.getDate().getDayOfMonth() < limitDay)
+                .toList();
+    }
+
+    public Map<AttendanceStatus, Integer> calculateAttendanceResult(final int limitDay) {
+        Map<AttendanceStatus, Integer> result = new HashMap<>();
+        result.put(AttendanceStatus.ATTENDANCE, 0);
+        result.put(AttendanceStatus.LATE, 0);
+        result.put(AttendanceStatus.ABSENCE, 0);
+
+        List<Attendance> attendances = getAllAttendances(limitDay);
+        for (Attendance attendance : attendances) {
+            AttendanceStatus status = attendance.getStatus();
+            if (status == AttendanceStatus.TRUANCY) {
+                status = AttendanceStatus.ABSENCE;
+            }
+            result.replace(status, result.get(status) + 1);
+        }
+        return result;
+    }
+
+    public int getLateCount(final int limitDay) {
+        List<Attendance> attendances = getAllAttendances(limitDay);
+        int count = 0;
+        for (Attendance attendance : attendances) {
+            if (attendance.getStatus() == AttendanceStatus.LATE) {
+                count++;
+            }
+        }
+        return count;
```

Comment:
```suggestion
        return (int) getAllAttendances(limitDay).stream()
                .filter(attendance -> attendance.getStatus() == AttendanceStatus.LATE)
                .count();
```
stream으로 변경가능할 것 같습니다.

Reply:
제안해주신 방법대로 하면 훨씬 코드가 깔끔해질 것 같아요! 다음에 비슷한 로직을 짤 때 꼭 참고하도록 하겠습니다 🙇🏻‍♀️

리팩터링 과정에서 로직이 수정되는 바람에, 제안해주신 방법은 코드에 반영하지 못했습니다 ㅜ

- 출석부가 **한 달 동안의 출석 내역만 담는다는 보장이 없기** 때문에, 전달받은 기한(시작 날짜 ~ 끝 날짜) 내에서 지각/결석 횟수를 계산할 수 있도록 변경
https://github.com/woowacourse/java-attendance/blob/48707fe657761d535e6aa6f02eda1ce53d47012e/src/main/java/domain/AttendanceBook.java#L57-L71

Reply:
코드를 다 살펴보고 난 뒤에 이 코멘트를 확인했네요! 여기도 Stream으로 변경이 가능합니다. 자세한 내용은 피드백 참고해주세요~